test/jasmine/tests/image_test.js

```diff
@@ -100,6 +101,82 @@ describe('image supplyDefaults', function() {
     });
 });
 
+describe('image smart layout defaults', function() {
+    var gd;
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    it('should reverse yaxis if only images are present', function() {
+        gd = {};
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}];
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.autorange).toBe('reversed');
+    });
+
+    it('should reverse yaxis even if another trace is present', function() {
+        gd = {};
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}, {type: 'scatter', y: [5, 3, 2]}];
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.autorange).toBe('reversed');
+    });
+
+    it('should NOT reverse yaxis if it\'s already defined', function() {
+        gd = {};
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}];
+        gd.layout = {yaxis: {autorange: false}};
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.autorange).toBe(false);
+    });
+
+    it('should set scaleanchor to make square pixels if only images are present', function() {
+        gd = {};
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}];
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.scaleanchor).toBe('x');
+    });
+
+    it('should set scaleanchor even if another trace is present', function() {
+        gd = {};
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}, {type: 'scatter', y: [5, 3, 2]}];
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.scaleanchor).toBe('x');
+    });
+
+    it('should NOT set scaleanchor if it\'s already defined', function() {
+        gd.data = [{type: 'image', z: [[[255, 0, 0]]]}];
+        gd.layout = {yaxis: {scaleanchor: 'x3'}};
+        supplyAllDefaults(gd);
+        expect(gd._fullLayout.yaxis.scaleanchor).toBe(undefined);
+    });
+
+    it('should constrain axes to domain if only images are present', function() {
```

Comment:
:hocho:  `only`

Reply:
Again, should I :hocho: the whole test?

Reply:
Done in https://github.com/plotly/plotly.js/pull/4307/commits/408b46ba5d1bddcd496b9397f1864b0749ecec89